### PR TITLE
Ranged Multi Attack Perk To Dao Conversion

### PR DESCRIPTION
**Pull request by Demojay**
Converted use of perks for determining ranged multi attack to use Dao mastery instead
Fixed damage functions error associated with throwing weapons and bow damage
Bonuses and references previously referring to redundant perks redirected to use Dao mastery instead.

### CODE REVIEW

File: classes/classes/Player.as [hunk 1]
```diff
@@ -1653,13 +1653,22 @@ use namespace CoC;
...
+		}
+		//Is Crossbow
+		public function isCrossbowTypeWeapon():Boolean
+		{
+			return weaponRangePerk == ItemConstants.WT_CROSSBOW;
 		}
 		//Is Thrown
 		public function isThrownTypeWeapon():Boolean
 		{
 			return weaponRangePerk == "Throwing";
```
Comment (jfmherokiller) [inline classes/classes/Player.as:1666]: change this to WT_THROWING for consistency sake

File: classes/classes/Scenes/Combat/Combat.as [hunk 2]
```diff
@@ -3151,16 +3159,15 @@ public class Combat extends BaseContent {
             if (player.hasStatusEffect(StatusEffects.ResonanceVolley)) outputText("Your bow nudges as you ready the next shot, helping you keep your aimed at [monster name].\n\n");
             multiArrowsStrike();
         }
-        if (player.weaponRangePerk == "Throwing"){
+        else if (player.weaponRangePerk == "Throwing"){
```
Comment (jfmherokiller) [inline classes/classes/Scenes/Combat/Combat.as:3134]: i think you can use isThrownTypeWeapon here

**Review by jfmherokiller — COMMENTED** [review]
some suggested changes

File: classes/classes/Player.as (hunk 1 above)
Comment (Demojay) [reply classes/classes/Player.as]: This has been changed to use the constant

File: classes/classes/Scenes/Combat/Combat.as (hunk 2 above)
Comment (Demojay) [reply classes/classes/Scenes/Combat/Combat.as]: This has been changed to use the method

**Comment by Ormael13** [comment]
public static const WT_THROWING:String      = "Throwing";
It was used for thrown melee weapons not thrown range weapons

**Comment by Demojay** [comment]
Thrown melee is "Thrown" while Thrown range is "Throwing".
You can see this in player.haveThrowableMeleeWeapon:

return (weaponClass("Thrown")) || hasAetherTwinsTierS1() || hasAetherTwinsTierS2();

while player.isThrownTypeWeapon is:

return weaponRangePerk == "Throwing";

**Pull request merged**